Zend/zend_API.h

```diff
@@ -1162,8 +1235,18 @@ static zend_always_inline int zend_parse_arg_str(zval *arg, zend_string **dest,
 	if (EXPECTED(Z_TYPE_P(arg) == IS_STRING)) {
 		*dest = Z_STR_P(arg);
 	} else if (EXPECTED(Z_TYPE_P(arg) < IS_STRING)) {
-		if (check_null && UNEXPECTED(Z_TYPE_P(arg) == IS_NULL)) {
-			*dest = NULL;
+		if (UNEXPECTED(Z_TYPE_P(arg) == IS_NULL)) {
+			if (check_null && (Z_TYPE_P(arg) == IS_NULL)) {
+				*dest = NULL;
+#if STH_DISABLE_NULL_TO_STRING	
+			} else {
+				return 0;
+#endif
```

Comment:
It seems I fixed it almost the same way before I saw your comment. I also found another small bug. Hope it's cleaner now. Thanks.